Tests/UnitTests/Networking/NetworkErrorTests.swift

```diff
@@ -0,0 +1,199 @@
+            ])
+
+        verifyPurchasesError(error,
+                             expectedCode: .invalidCredentialsError,
+                             underlyingError: underlyingError,
+                             userInfoKeys: [Backend.RCAttributeErrorsKey])
+    }
+
+}
+
+class NetworkErrorTests: XCTestCase {
+
+    func testSuccessfullySyncedTrue() {
+        let errors = [
+            error(Self.responseError(.invalidRequest)),
+            error(Self.responseError(.notModified)),
+            error(Self.responseError(.success))
+        ]
+
+        for error in errors {
+            check(error.0.successfullySynced,
+                  condition: beTrue(),
+                  descrition: "Expected error to be successfully synced",
+                  file: error.1,
+                  line: error.2)
+        }
+    }
+
+    func testSuccessfullySyncedFalse() {
+        let errors = [
+            error(Self.decodingError),
+            error(Self.networkError),
+            error(Self.dnsError),
+            error(Self.unableToCreateRequestError),
+            error(Self.unexpectedResponseError),
+            error(Self.responseError(.notFoundError)),
+            error(Self.responseError(.internalServerError))
+        ]
+
+        for error in errors {
+            check(error.0.successfullySynced,
+                  condition: beFalse(),
+                  descrition: "Expected error to be not successfully synced",
+                  file: error.1,
+                  line: error.2)
+        }
+    }
+
+    func testFinishableTrue() {
+        let errors = [
+            error(Self.responseError(.invalidRequest)),
+            error(Self.responseError(.notFoundError)),
+            error(Self.responseError(.notModified)),
+            error(Self.responseError(.success))
+        ]
+
+        for error in errors {
+            check(error.0.finishable,
+                  condition: beTrue(),
+                  descrition: "Expected error to be finishable",
+                  file: error.1,
+                  line: error.2)
+        }
+    }
+
+    func testFinishableFalse() {
+        let errors = [
+            error(Self.decodingError),
+            error(Self.networkError),
+            error(Self.dnsError),
+            error(Self.unableToCreateRequestError),
+            error(Self.unexpectedResponseError),
+            error(Self.responseError(.internalServerError))
+        ]
+
+        for error in errors {
+            check(error.0.finishable,
+                  condition: beFalse(),
+                  descrition: "Expected error to not be finishable",
+                  file: error.1,
+                  line: error.2)
+        }
+    }
+
+    // MARK: - Helpers
+
+    private func error(
+        _ error: NetworkError,
+        file: FileString = #file, line: UInt = #line
+    ) -> (NetworkError, FileString, UInt) {
+        return (error, file, line)
+    }
```

Comment:
This is simply so that Xcode shows up the test failure from the line that creates the error, making it much easier to see which one is failing from the array.

Reply:
👍🏻